python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -109,6 +109,10 @@ class AutoscalerSummary:
         default_factory=lambda: NodeAvailabilitySummary({})
     )
     pending_resources: Dict[str, int] = field(default_factory=lambda: {})
+    # A mapping from node name (the same key as `usage_by_node`) to node type.
+    # Optional for deployment modes which have the concept of node types and
+    # backwards compatibility.
+    node_type_mapping: Optional[Dict[str, str]] = None
```

Comment:
Would be good for the name of the map to also reflect the key of the map, but I guess we don't have a good handle on what the key is?
(`ip_to_node_type` maybe for this map?)